catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -403,7 +410,11 @@ private ContentItem readContent(URI uri) throws StorageException {
     if (REF_EXT.equals(extension)) {
       extension = FilenameUtils.getExtension(FilenameUtils.removeExtension(filename));
       try {
-        reference = new URI(new String(Files.readAllBytes(file), Charset.forName("UTF-8")));
+        reference =
```

Comment:
I was a little confused on the first read-through. It's not the new code; it is the old code. The variable `file` is used for two different things. In the first instance, it is a file that contains a reference to the actual content item. Then the variable is reused as part of getting the bytes of the item itself. When I first looked at the new code and saw `readAllBytes` I thought it was trying to read the actual content! I knew that couldn't be because we would blow out the heap if the content were big enough. 

I recommend a small refactoring to use two different variables for the two different uses.

Reply:
Yeah, the code you mentioned is pretty confusing.

The problem with trying to refactor this into two methods is that we don't know whether or not the URI we pass in from the `UpdateStorageRequest` or `ReadStorageRequest` is an "external reference" or not ([see this comment](https://github.com/codice/ddf/pull/4220/files#diff-9b68a9cb68bedc7e0609dcfdd52cf3dfL401)).

That means that that check would either have to be done either in the current method or pulled out into the calling methods. I personally think it is cleaner to have the additional check in the current method in order to avoid duplicated code.

Let me know what you think of my analysis. And do note that I plan on refactoring this method a little to increase readability.